src/video/cocoa/cocoa_wnd.mm

```diff
@@ -132,6 +161,37 @@ static NSUInteger CountUtf16Units(const char *from, const char *to)
 	return unicode_str;
 }
 
+static void CGDataFreeCallback(void *, const void *data, size_t)
+{
+	delete[] (const uint32 *)data;
+}
+
+/**
+ * Render an OTTD sprite to a Cocoa image.
+ * @param sprite_id Sprite to make a NSImage from.
+ * @param zoom Zoom level to render the sprite in.
+ * @return Autorelease'd image or nullptr on any error.
+ */
+static NSImage *NSImageFromSprite(SpriteID sprite_id, ZoomLevel zoom)
+{
+	if (!SpriteExists(sprite_id)) return nullptr;
+
+	/* Fetch the sprite and create a new bitmap */
+	Dimension dim = GetSpriteSize(sprite_id, nullptr, zoom);
+	std::unique_ptr<uint32[]> buffer = DrawSpriteToRgbaBuffer(sprite_id, zoom);
+	if (!buffer) return nullptr; // failed to blit sprite or we're using an 8bpp blitter.
```

Comment:
Is this comment still true? As you put a lot of code to render 8bpp in the function?